### PR TITLE
Issue702: Add example 03SnakeChargeScan

### PR DESCRIPTION
**Pull request by BrllsRX**
Complete the snaked charge scan as an example of:
Task 1: explore functionalities of `qupulse.pulses.plotting.plot` and piecewise the pulse artificially.
Task 2: simplify the pulse by decoupling the measurement windows from the piecewise pulse.
Task 3: testing helper functions `with_*` as mentioned in issue [710](https://github.com/qutech/qupulse/issues/710)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,6 @@ MATLAB/+qc/personalPaths.mat
 .idea/
 .mypy_cache/*
 tests/hardware/WX2184C.exe
+research_tech/*
```
Comment (terrorfisch) [inline .gitignore:17]: Please do not put personal files here.
`.vscode` is fine but the other two aren't. You can either not commit them or insert these lines into `.git/info/exclude`

**Comment by terrorfisch** [comment]
Can you rebase on the current master?

**Comment by terrorfisch** [comment]
The parameter constraints examples is supposed to start with "01"

**Comment by terrorfisch** [comment]
My idea was to group the examples in 
00: PulseTemplates
01: Additional functionality
02: reserved for future use
03: Real-world examples

The snake case should go to 03 I guess.

**Comment by BrllsRX** [comment]
alright. It is now renamed and I will push my local change soon. Let me know if any problems remain.

**Comment by terrorfisch** [comment]
Looks good. Can you merge the master and reexecute the parameter contraints example? It looks like the current is based on an old master branch version with issue #707 unfixed (is fixed in current master).

**Comment by terrorfisch** [comment]
Rebased in #737

**Pull request closed without merging**